urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlab/examples/TableauState.java

```diff
@@ -0,0 +1,523 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package permlab.examples;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Random;
+import permlib.PermUtilities;
+
+/**
+ *
+ * @author MichaelAlbert
+ */
+public class TableauState {
+
+    private static int inversions(int[] p) {
+        int result = 0;
+        for (int i = 0; i < p.length; i++) {
+            for (int j = i + 1; j < p.length; j++) {
+                if (p[i] > p[j]) {
+                    result++;
+                }
+            }
+        }
+        return result;
+    }
+
+    private static void checkDeletions(int[] newElements) {
+        int[] deletedCopy = Arrays.copyOf(newElements, newElements.length - 1);
+        boolean hasBadDeletion = false;
+        for (int deletedElement = newElements.length - 1; deletedElement >= 0; deletedElement--) {
+//            System.out.println("Checking deletion " + Arrays.toString(deletedCopy));
+            hasBadDeletion = !isMultiplicityFree(deletedCopy);
+            if (hasBadDeletion || deletedElement == 0) {
+                break;
+            }
+            deletedCopy[deletedElement - 1] = newElements[deletedElement];
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 34:41, 485:491.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.